docs/deploy-on-k8s.md

```diff
@@ -0,0 +1,115 @@
+
+```sh
+kubectl get pods -n oms
+```
+
+You should see something like the following:
+
+```
+NAME                              READY   STATUS    RESTARTS   AGE
+billing-api-0                     1/1     Running   0          24h
+billing-worker-5498c6ffd4-klfwk   1/1     Running   0          24h
+codec-server-847b59b84f-9ct54     1/1     Running   0          24h
+main-api-0                        1/1     Running   0          24h
+main-worker-777c59ccdb-jx46w      1/1     Running   0          24h
+web-557c8dc97-p5fkq               1/1     Running   0          24h
+```
+
+Once the READY column shows "1/1" for all the pods, the application is up.
+
+Please note that while in most regards the deployment is a standard Kubernetes setup, there is one thing which is a little unusual. Our API services make use of a cache to ensure that they can serve the listings for orders and shipments quickly. This cache lives in an SQLite file on disk, which makes the API services stateful. If an API pod needs to be restarted (due to upgrade, node failure etc) then it must be brought back up with the same disk as before to ensure it does not lose it's cache. For this reason, the API services use Statefulset instead of Deployment. The cache is mounted via a Kubernetes volume to ensure that the pod will maintain it's disk between runs.
+
+As this cache uses the disk, which is not easily shared amongst pods, this also means the API Statefulsets cannot be safely scaled. If there were to be more than one main-api pod, for example, each pod would see different order creations, and therefore have a different cache of orders. This would result in unpredictable behaviour.
+
+These are all limitations of our deliberately naeive caching mechanism, and would not occur should the APIs use a shared database such as MySQL or Postgres.
+
+## Using the application
+
+So now you have the application installed in Kubernetes, how can you use it? The standard practise for accessing a service inside a Kubernetes cluster would be to create a Service with type LoadBalancer. On AWS, for example, this would cause AWS to automatically provision an ELB or ALB for you and connect it to the Kubernetes service, allowing external access. As we are only providing a demonstration system and don't wish to make things available over the public internet, we are not using LoadBalancer in our deployment. Instead we provide some scripts which use Kubernetes port-forwarding so that you can reach the Services in Kubernetes from your local machine.
+
+To see the application's Web UI you can run the script:
+
+```sh
+./deployments/port-forward-web.sh
+```
+
+That command will continue to run until you kill it (with Ctrl+C) for example. While the command is running it will forward any traffic from your localmachine on port 3000 to the application's Web UI in the cluster. You should now be able to see the application at: http://localhost:3000/
```

Comment:
```suggestion
That command will continue to run until you terminate it (with Ctrl+C) for example. While the command is running it will forward any traffic from your local machine on port 3000 to the application's Web UI in the cluster. You should now be able to see the application at: http://localhost:3000/
```